jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8307,13 +8307,71 @@ public String resolveConstantDesc(MethodHandles.Lookup lookup) {
 	}
 	
 	/**
-	 * Apply a function to this string. The function expects a single String input and returns an R.
+	 * Apply a function to this string. The function expects a single String input
+	 * and returns an R.
 	 * 
-	 * @param f - the functional interface to be applied
+	 * @param f 
+	 *			  the functional interface to be applied
+	 *
 	 * @return the result of application of the function to this string
+	 * 
+	 * @since 12
 	 */
 	public <R> R transform(Function<? super String, ? extends R> f) {
 		return f.apply(this);
 	}
+
+	/**
+	 * Indents each line of the string depending on the value of n, and normalizes
+	 * line terminators to the newline character "\n".
+	 * 
+	 * @param n 
+	 *			  the number of spaces to indent the string
+	 *
+	 * @return the indented string with normalized line terminators
+	 * 
+	 * @since 12
+	 */
+	public String indent(int n) {
+		Stream<String> lines = lines();
+		Iterator<String> iter = lines.iterator();
+		StringBuilder builder = new StringBuilder();
+
+		while (iter.hasNext()) {
+			String currentLine = iter.next();
+
+			if (n > 0) {
+				String spaces = " ".repeat(n);
+				currentLine = spaces + currentLine;
+			} else if (n < 0) {
+				int absN = Math.abs(n);
+				if (currentLine.isBlank()) {
```

Comment:
Does this need to know if the line is blank?  checking for a blank line is equivalent to the `Character.isWhitespace(currentLine.charAt(start))` loop below.  I think the space case can be removed and handled by the substring() call